src/corona_hakab_model/supervisor.py

```diff
@@ -34,129 +34,41 @@ def snapshot(self, manager):
         for s in self.supervisables:
             s.snapshot(manager)
 
+    def dump(self):
+        output_folder = os.path.join(os.path.split(os.path.dirname(os.path.realpath(__file__)))[0], "output")
+
+        with open(os.path.join(output_folder, datetime.now().strftime("%Y%m%d-%H%M%S") + ".csv"), 'w',
+                  newline="") as output_file:
+            published_data = [v.publish() for v in self.supervisables]
+            csv_writer = csv.writer(output_file)
+            # write data rows
+            for i in range(self.manager.consts.total_steps + 1):
+                row = [i]
+                if i == 0:  # header row
+                    row = ["days"]
+                for data in published_data:
+                    # first column is always the #day
+                    row = row + data[i][1:]
```

Comment:
isn't this gonna attach real data to the header row as well?

Reply:
i think that the first row of each supervisable is its name, which should be in the header row

Reply:
Anyway I'm switching most of this code to use Pandas